file/file-api/src/main/java/uk/gov/justice/services/file/api/requester/FileRequester.java

```diff
@@ -7,6 +7,14 @@
  */
 public interface FileRequester {
 
-    Optional<byte[]> request(final String fileId, final String fileMimeType, final String fileName);
+    /**
+     * Requests a file from the FileService.
+     * @param fileId - the unique id of the file.
+     * @param fileMimeType - mime-type of the file.
+     * @param fileName - name of the file.
+     * @param stream - true to get files in stream mode, else false.
+     * @return content of the file in byte[].
+     */
+    Optional<byte[]> request(final String fileId, final String fileMimeType, final String fileName, final boolean stream);
```

Comment:
Okay, so as mentioned the other day.. maybe run this by David and see his opinion...

Would it be cleaner to have a FileRequester and a SteamingFileRequester interface (Which extends the FileRequester) ?

It means if we have a file service that can't do streaming (and we know it won't be needed to) we can use the simplified method rather than having to pass false each time.